### PR TITLE
ref(app-platform): Remove service hooks on app uninstall

### PR DESCRIPTION
**Pull request by MeredithAnya**
When an app is uninstalled we want to remove all the service hooks associated with the install.

### CODE REVIEW

File: src/sentry/mediators/sentry_app_installations/destroyer.py [hunk 1]
```diff
@@ -18,5 +21,15 @@ def _destroy_authorization(self):
     def _destroy_grant(self):
         self.install.api_grant.delete()
 
+    def _destroy_service_hooks(self):
+        hooks = ServiceHook.objects.filter(
+            application=self.install.sentry_app.application,
```
Comment (mnoble) [inline src/sentry/mediators/sentry_app_installations/destroyer.py]: You can avoid the `application` lookup here by doing:

```python
application_id=self.install.sentry_app.application_id
```

File: src/sentry/mediators/sentry_app_installations/destroyer.py [hunk 2]
```diff
@@ -18,5 +21,15 @@ def _destroy_authorization(self):
...
+            actor_id=self.install.id,
+        )
```
Comment (mnoble) [inline src/sentry/mediators/sentry_app_installations/destroyer.py]: Might be useful to add a `SentryAppInstallation#service_hooks` method that exposes these. Not a blocker – just a thought.

File: src/sentry/mediators/sentry_app_installations/destroyer.py [hunk 3]
```diff
@@ -18,5 +21,15 @@ def _destroy_authorization(self):
...
+        for hook in hooks:
+            service_hooks.Destroyer.run(
+                service_hook=hook,
+            )
```
Comment (mnoble) [inline src/sentry/mediators/sentry_app_installations/destroyer.py]: nit: you don't need to break this up onto three lines.

File: tests/sentry/mediators/sentry_app_installations/test_destroyer.py [hunk 4]
```diff
@@ -40,6 +42,16 @@ def test_deletes_grant(self):
 
         assert not ApiGrant.objects.filter(pk=grant.id).exists()
 
+    def test_deletes_service_hooks(self):
+        hook = ServiceHook.objects.get(
+            application=self.sentry_app.application,
+            actor_id=self.install.id,
+        )
```
Comment (mnoble) [inline tests/sentry/mediators/sentry_app_installations/test_destroyer.py]: Just fyi, there's a `create_service_hook()` function.

**Review by mnoble — COMMENTED** [review]
Looks great! Only change I'd argue should happen before merge is the `application_id` one. The rest are just suggestions or thoughts.

File: tests/sentry/mediators/sentry_app_installations/test_destroyer.py (hunk 4 above)
Comment (MeredithAnya) [reply tests/sentry/mediators/sentry_app_installations/test_destroyer.py]: looking at it, shouldn't the actor be the install (in the case of sentry apps)? right now the `create_service_hook` creates a user to be the actor

**Review by mnoble — APPROVED** [review]
🎉👍

**Pull request merged**